src/components/webview/MainWebviewProvider.ts

```diff
@@ -0,0 +1,501 @@
+import {
+	WebviewViewProvider,
+	WebviewView,
+	Uri,
+	ExtensionContext,
+	workspace,
+	commands,
+} from 'vscode';
+import { Message, MessageBus, MessageKind } from '../messageBus';
+import {
+	TreeNode,
+	View,
+	WebviewMessage,
+	WebviewResponse,
+} from './webviewEvents';
+import { WebviewResolver } from './WebviewResolver';
+import {
+	CaseElement,
+	Element,
+	ElementHash,
+	FileElement,
+	RootElement,
+} from '../../elements/types';
+import { Job, JobHash, JobKind } from '../../jobs/types';
+import { debounce } from '../../utilities';
+import { JobManager } from '../jobManager';
+import { CaseHash, CaseWithJobHashes } from '../../cases/types';
+import {
+	buildJobElement,
+	compareJobElements,
+} from '../../elements/buildJobElement';
+import {
+	buildFileElement,
+	compareFileElements,
+} from '../../elements/buildFileElement';
+import {
+	buildCaseElement,
+	compareCaseElements,
+} from '../../elements/buildCaseElement';
+import { CaseManager } from '../../cases/caseManager';
+
+const getElementIconBaseName = (kind: Element['kind']): string => {
+	switch (kind) {
+		case 'CASE':
+			return 'case.svg';
+		case 'FILE':
+			return 'ts2.svg';
+		default:
+			return 'bluelightbulb.svg';
+	}
+};
+
+export const ROOT_ELEMENT_HASH: ElementHash = '' as ElementHash;
+
+export class IntuitaProvider implements WebviewViewProvider {
```

Comment:
can you leave a TODO to split this file?

Reply:
actually there are a lot of opportunities for refactoring in this file. I've already planned multiple things for the next weekends